__tests__/app.test.js

```diff
@@ -433,3 +464,98 @@ describe("GET /api/articles/:article_id/comments", () => {
     });
   });
 });
+
+describe("POST api/articles/:article_id/comments", () => {
+  test("200: returns object of article with updated votes", () => {
+    return request(app)
+      .post(`/api/articles/1/comment`)
+      .send({
+        username: "icellusedkars",
+        body: "blah blah blah blah something newsy blah blah blah",
+      })
+      .expect(200)
+      .then(({ body }) => {
+        const comment = body.comment;
+
+        expect(comment).toEqual({
+          comment_id: expect.any(Number),
+          article_id: 1,
+          author: "icellusedkars",
+          body: "blah blah blah blah something newsy blah blah blah",
+          created_at: expect.any(String),
+          votes: 0,
+        });
+      });
+  });
+
+  describe("404: article id doesn't exist", () => {
+    test("404: returns message, article id doesnt exist", () => {
+      return request(app)
+        .post(`/api/articles/13/comment`)
+        .send({
+          username: "icellusedkars",
+          body: "blah blah blah blah something newsy blah blah blah",
+        })
+        .expect(404)
+        .then(({ body }) => {
+          expect(body).toEqual({
+            status: 404,
+            message: "article id doesn't exist",
+          });
+        });
+    });
+  });
+
+  describe("400: article id is wrong type", () => {
+    test("400: returns message of bad request", () => {
+      return request(app)
+        .post(`/api/articles/number/comment`)
+        .send({
+          username: "icellusedkars",
+          body: "blah blah blah blah something newsy blah blah blah",
+        })
+        .expect(400)
+        .then(({ body }) => {
+          expect(body).toEqual({
+            status: 400,
+            message: "Bad Request",
+          });
+        });
+    });
+  });
+  describe("400: username in body doesn't exist", () => {
+    test("400: returns message of bad request", () => {
+      return request(app)
+        .post(`/api/articles/number/comment`)
+        .send({
+          username: "somebodyanybody",
+          body: "blah blah blah blah something newsy blah blah blah",
+        })
+        .expect(400)
```

Comment:
I don't think that this is a bad request as the request body is properly formed. I'd go for a 404 here as the user is not found.